ext/third_party/zi_lib/zi/mesh/marching_cubes.hpp

```diff
@@ -160,6 +160,19 @@ class marching_cubes: non_copyable
         num_faces_ = 0;
     }
 
+    bool erase(const Type& t) 
+    {
+        try {
+            std::vector< triangle > &tri = meshes_.at(t);
+            std::size_t num_erased = meshes_.erase(t);
+            num_faces_ -= tri.size();
```

Comment:
`tri` is not valid anymore after you erase it. Maybe just copy the size of the vector `meshes_.at(t).size()` further above, instead of keeping a reference.